lambdas/ec2/schema_enforcer.py

```diff
@@ -0,0 +1,152 @@
+        raise Exception('EC2 instance deleted because no termination_date found within 4 minutes of creation.')
+
+
+def validate_ec2_termination_date(ec2_instance):
+    """
+    :param ec2_instance: a boto3 resource representing an Amazon EC2 Instance.
+
+    Validates that an ec2 instance has a termination_date in the future.
+    """
+    ec2_termination_date = check_for_tag(ec2_instance, 'termination_date')
+    if ec2_termination_date is False:
+        print("No termination_date found; deleting ec2 instance and raising Exception.")
+        # TODO: Remove this code whence the reaper goes live
+        # ec2_instance.delete()
+        raise ValueError('No termination_date found; ec2 instance deleted.')
+    try:
+        dateutil.parser.parse(ec2_termination_date) - timenow_with_utc()
+    except Exception as e:
+        if e is TypeError:
+            if re.search(r'(offset-naive).+(offset-aware)', e.__str__):
+                print('The termination_date requires a UTC offset; terminating instance')
+            else:
+                print('Unable to parse the termination_date; deleting ec2 instance and raising original exception.')
+        # TODO: Remove this comment whence the reaper goes live
+        # ec2_instance.delete()
+        raise
+
+    if dateutil.parser.parse(ec2_termination_date) > timenow_with_utc():
+        ttl = dateutil.parser.parse(ec2_termination_date) - timenow_with_utc()
+        print("EC2 instance will be terminated {0} seconds from now, roughly".format(ttl.seconds))
+    else:
+        print("The termination_date has passed; deleting ec2 resource and raising exception")
+        # TODO: This code should also eventually report terminated instances
+        # to some log somewhere.
+        # ec2_instance.terminate()
+        raise Exception("The EC2 instance has been deleted because the termination_date has passed.")
+
+
+
+def calculate_lifetime_delta(lifetime_value):
+    """
+    :param lifetime_value: string that with an integer and a single letter unit of (w)eeks, (d)ays, or (h)ours.
+
+    Takes a string of of an integer value with a 1 letter unit of w(weeks), d(days), h(hours).
+    """
+    regex = r'^([0-9]+)(w|d|h)$'
+    match = re.search(regex, lifetime_value)
+    if match is None:
+        print("Invalid lifetime syntax; please provide valid integer followed by 1 letter unit: 1w=1week, 2d=2day, 3h=3hours.")
+        # TODO: Remove this comment whence the reaper goes live
+        # ec2_instance.delete()
+        raise ValueError("The lifetime value {0} cannot be parsed".format(lifetime_value))
+
+    toople = match.groups()
+    unit = toople[1]
+    length = int(toople[0])
+    if unit == 'w':
+        return datetime.timedelta(weeks=length)
+    elif unit =='h':
+        return datetime.timedelta(hours=length)
+    elif unit == 'd':
+        return datetime.timedelta(days=length)
+    else:
+        raise ValueError("Unable to parse the unit '{0}'".format(unit))
+
+ec2 = boto3.resource('ec2')
```

Comment:
This is the assignment of the boto3 abstraction for ec2; I wasn't sure if there was functionality that should exist with this outside of the lambda handler, so I left it here in the global scope of the script in case it was needed outside of the handler.